src/components/TodoComponents/TodoForm.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+class TodoForm extends React.Component {
+    constructor(){
+        super();
+        this.state = {
+            itemInput: ''
+        }
+    }
+
+    handleInput = e => {
+        this.setState({itemInput: e.target.value});
+    }
+
+    handleSubmit = e => {
+        e.preventDefault();
+        this.props.add({task: this.state.itemInput, 
+                        id: Date.now(),
+                        completed: false})
+        this.setState({itemInput: ''});
+    }
+
+    handleClick = () => {
+        this.props.clear();
+    }
```

Comment:
I think it's important to bring up the semantic naming thing one more time, the only reason I could tell what this function might be doing was from reading through it, `handleClearButtonClick` would tell me a lot more about it with making me look through the code and seeing what button you set it on and trying to follow your perspective with little reference.